kv/label.go

```diff
@@ -85,7 +85,8 @@ func (s *Service) findLabelByID(ctx context.Context, tx Tx, id influxdb.ID) (*in
 
 func filterLabelsFn(filter influxdb.LabelFilter) func(l *influxdb.Label) bool {
 	return func(label *influxdb.Label) bool {
-		return (filter.Name == "" || (filter.Name == label.Name))
+		return (filter.Name == "" || (filter.Name == label.Name) &&
+			filter.OrganizationID == nil || (*filter.OrganizationID == label.OrganizationID))
```

Comment:
If filter.Name != label.Name && filter.OrganizationID == nil, this might fail. To be safe add a check with filter.ID!=nil &&(*filter.OrganizationID == label.OrganizationID)

Reply:
Yep. Tests caught this too, was causing a segfault.